dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileBig.cpp

```diff
@@ -140,6 +138,23 @@ ColumnFilePersistedPtr ColumnFileBig::createFromCheckpoint(
     return std::shared_ptr<ColumnFileBig>(dp_file);
 }
 
+ColumnFilePersistedPtr ColumnFileBig::createFromCheckpoint(
+    DMContext & dm_context,
+    const RowKeyRange & target_range,
+    const dtpb::ColumnFileBig & cf_pb,
+    UniversalPageStoragePtr temp_ps,
+    WriteBatches & wbs)
+{
+    UInt64 file_page_id = cf_pb.id();
+    size_t valid_rows = cf_pb.valid_rows();
+    size_t valid_bytes = cf_pb.valid_bytes();
+
+    auto remote_data_store = dm_context.global_context.getSharedContextDisagg()->remote_data_store;
+    auto dmfile = restoreDMFileFromCheckpoint(dm_context, remote_data_store, temp_ps, wbs, file_page_id);
+    auto * dp_file = new ColumnFileBig(dmfile, valid_rows, valid_bytes, target_range);
```

Comment:
Why don't we use make_shared here?

Reply:
private constructor

Reply:
Why not use a static function?
```
struct A {
private:
    A() {}
public:
    static std::shared_ptr<A> makeA() {
        return std::make_shared<A>();
    }  
};
```

Reply:
It is just the same as public constructor?

```cpp
struct A {
public:
    A() {}
};
```

Reply:
OK... May be I should firstly ask the question that why the constructor is private?

Reply:
IMHO, it is used to make sure the stat (`valid_rows`, `valid_bytes`, and `segment_range`) is right.

Reply:
the public constructor will calculate stat:

```cpp
ColumnFileBig::ColumnFileBig(const DMContext & context, const DMFilePtr & file_, const RowKeyRange & segment_range_)
    : file(file_)
    , segment_range(segment_range_)
{
    calculateStat(context);
}

void ColumnFileBig::calculateStat(const DMContext & dm_context)
{
    auto index_cache = dm_context.global_context.getMinMaxIndexCache();

    auto pack_filter = DMFilePackFilter::loadFrom(
        file,
        index_cache,
        /*set_cache_if_miss*/ false,
        {segment_range},
        EMPTY_RS_OPERATOR,
        {},
        dm_context.global_context.getFileProvider(),
        dm_context.getReadLimiter(),
        dm_context.scan_context,
        /*tracing_id*/ dm_context.tracing_id,
        ReadTag::Internal);

    std::tie(valid_rows, valid_bytes) = pack_filter.validRowsAndBytes();
}
```

Reply:
Alright, then this issue is better not to be addressed on this PR.